pkg/operator/sync.go

```diff
@@ -244,6 +246,98 @@ func (optr *Operator) syncRenderConfig(_ *renderConfig) error {
 		return fmt.Errorf("refusing to read images.json version %q, operator version %q", imgs.ReleaseVersion, optrVersion)
 	}
 
+	// handle image registry certificates.
+	// parse these, add them to ctrlcfgspec and then handle these in the daemon write to disk function
+	cfg, err := optr.imgLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+	imgRegistryUsrData := []v1.ImageRegistryBundle{}
+	if cfg.Spec.AdditionalTrustedCA.Name != "" {
+		klog.Infof("got image.config.openshift.io/cluster, additionalCA: %s", cfg.Spec.AdditionalTrustedCA.Name)
+		cm, err := optr.clusterCmLister.ConfigMaps("openshift-config").Get(cfg.Spec.AdditionalTrustedCA.Name)
+		if err != nil {
+			klog.Warningf("could not find configmap specified in image.config.openshift.io/cluster with the name %s", cfg.Spec.AdditionalTrustedCA.Name)
+		} else {
+			for key, d := range cm.Data {
+				klog.Infof("user provided data for img registry, key: %s data: %s", key, cm.Data[key])
+				raw, err := base64.StdEncoding.DecodeString(d)
+				if err != nil {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: []byte(d),
+					})
+				} else {
+					imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+						File: key,
+						Data: raw,
+					})
+				}
+			}
+			for key, bd := range cm.BinaryData {
+				klog.Infof("data for img registry, key: %s data: %s", key, cm.Data[key])
+				imgRegistryUsrData = append(imgRegistryUsrData, v1.ImageRegistryBundle{
+					File: key,
+					Data: bd,
+				})
+			}
+		}
+	}
+
+	imgRegistryData := []v1.ImageRegistryBundle{}
+	cm, err := optr.clusterCmLister.ConfigMaps("openshift-image-registry").Get("image-registry-certificates")
+	if err != nil {
+		klog.Info("the image registry is not installed so the MCO cannot manage the image-registry-certificates configmap.")
```

Comment:
Similar to above, should this error be more fatal?

Reply:
actually, this is wrong. This should be the openshift-config-managed configmap. I am not sure what the expected behavior is here... if the CIRO does not exist, will the configmap still exist?